module/zfs/dsl_scan.c

```diff
@@ -403,6 +406,94 @@ dsl_scan_cancel(dsl_pool_t *dp)
 	    dsl_scan_cancel_sync, NULL, 3, ZFS_SPACE_CHECK_RESERVED));
 }
 
+boolean_t
+dsl_scan_is_paused_scrub(const dsl_scan_t *scn)
+{
+	if (dsl_scan_scrubbing(scn->scn_dp) &&
+	    scn->scn_phys.scn_flags & DSF_SCRUB_PAUSED)
+		return (B_TRUE);
+
+	return (B_FALSE);
+}
+
+static int
+dsl_scrub_pause_resume_check(void *arg, dmu_tx_t *tx)
+{
+	pool_scrub_cmd_t *cmd = arg;
+	dsl_pool_t *dp = dmu_tx_pool(tx);
+	dsl_scan_t *scn = dp->dp_scan;
+
+	ASSERT(*cmd == POOL_SCRUB_NORMAL || *cmd == POOL_SCRUB_PAUSE);
```

Comment:
**nit**: I know that all the callers verify that `cmd` is valid before calling this so the ASSERT is safe.  However, since the purpose of the `_check()` function as described in `dsl_synctask.c` is to determine exactly this kind of thing it would make sense to return ENOTSUP for an invalid cmd rather that ASSERT.  The check in `spa_scrub_pause_resume()` could then be removed.  Since this isn't ever going to be a hot code path it's not critial that calling this via `dsl_sync_task` is a bit more expensive.

How about

```c
        pool_scrub_cmd_t cmd = *(pool_scrub_cmd_t *)arg;

        if (cmd == POOL_SCRUB_PAUSE) {
                ...
        } else if (cmd == POOL_SCRUB_NORMAL) {
                ...
        } else {
                return (SET_ERROR(ENOTSUP));
        }
```